Dictionaries/hrv_OCRFixReplaceList.xml

```diff
@@ -950,433 +906,440 @@
+    <RegEx find="suštin(e|i|om)" replaceWith="biti" />
+    <RegEx find="([sS])vat(am|aš|a|amo|ate|aju|ajući|ali|ale|alo|aše)" replaceWith="$1hvać$2" />
+    <RegEx find="([sS])vedo(čim|čiš|či|čimo|čite|če|k|ke|ka|ku|kom|ci|cima|kinja|kinju|kinji|čanstvo|čanstva|čili|čio|čila|čiti)" replaceWith="$1vjedo$2" />
+    <RegEx find="([sS])vesn(a|e|i|o|u|om|oj|ima|ost|ošću)" replaceWith="$1vjesn$2" />
     <!-- razlikuju se svjetlo i svijetlo no tu automatske pomoći nema, već je na korisnicima da dodaju i gdje je potrebno! -->
-    <RegEx find="(s|S)vetl(a|e|i|o|u|om|ost|osti|ošću)" replaceWith="$1vjetl$3" />
-    <RegEx find="(s|S)vetsk(a|e|i|o|u|om|og|oj)" replaceWith="$1vjetsk$2" />
-    <RegEx find="(s|S)veštenik(a|e|u|om)?" replaceWith="$1većenik$2" />
-    <RegEx find="(s|S)veštenic(a|e|u|om|i|ima)" replaceWith="$1većenic$2" />
-    <RegEx find="(s|S)vež(a|e|u|om|oj)?" replaceWith="$1vjež$2" />
-    <RegEx find="(s|S)vide(la|li|lo)" replaceWith="$1vidje$2" />
-    <RegEx find="(s|S)vjet(a|u|om)?" replaceWith="$1vijet$2" />
-    <RegEx find="(s|S)vijet(ski|skim|sku|skog|skom)" replaceWith="$1vjet$2" />
-    <RegEx find="(Španij)(a|e|u|om)" replaceWith="Španjolsk$2" />
-    <RegEx find="(Špansk)(a|e|i|o|u|oj|om|im)" replaceWith="Španjolsk$2" />
-    <RegEx find="(takmičenj)(a|e|u|ima|em)" replaceWith="natjecanj$2" />
-    <RegEx find="\b([tT])(ač)(an|na|no|nom|noj|nog|nima|niji|nijim|nije|nijem|niju|nijoj)" replaceWith="$1oč$3" />
-    <RegEx find="(talentov)(an|ana|ani|ano|anom|anoj|anima)" replaceWith="talentir$2" />
-    <RegEx find="\b(t|T)(el)(o|a|u|om|ima)" replaceWith="$1ijel$3" />
-    <RegEx find="\b([tT])(era)(m|š|a|mo|te|ju|ti|la|o|li|še)?\b" replaceWith="$1jera$3" />
-    <RegEx find="\b([tT])(esn)(a|e|i|o|u|om|oj|og|ima)" replaceWith="$1ijesn$3" />
-    <RegEx find="\b([tT])(oleris)(ao|ala|ale|ali|ati)" replaceWith="$1olerir$3" />
-    <RegEx find="(u|U)(bj?edi)(o|m|š|mo|te|ti|la|le|li|lo|vši)?" replaceWith="$1vjeri$3" />
-    <RegEx find="([uU])(bedljivo)(st|šću)?" replaceWith="$1vjerljivo$3" />
-    <RegEx find="([uU])(beđen)(a|e|i|o|u|oj|om|ima|ost|ošću)?" replaceWith="$1vjeren$3" />
-    <RegEx find="\b([uU])(bic)(a|e|i|o|u|om|ima)" replaceWith="$1bojic$3" />
-    <RegEx find="\b([uU])(bistv)(a|o|u|om|ima)" replaceWith="$1bojstv$3" />
-    <RegEx find="(učestvuje)(m|š|mo|te)?" replaceWith="sudjeluje$2" />
-    <RegEx find="([uU])(cen)(a|e|i|u|om|ama|nu|om)" replaceWith="$1cjen$3" />
-    <RegEx find="(učestvo)(vala|vao|valo|vali|vale|vati|vanje|vanjem)" replaceWith="sudjelo$2" />
-    <RegEx find="\b(ulepša)(m|š|mo|te|ti|ju|la|le|li|lo|vam|vaš|vamo|vate|vaju|vati|vaše|nom|nim|noj|nima)?" replaceWith="uljepša$2" />
+    <RegEx find="([sS])vetl(a|e|i|o|u|om|ost|osti|ošću)" replaceWith="$1vjetl$2" />
+    <RegEx find="([sS])veštenik(a|e|u|om)?" replaceWith="$1većenik$2" />
+    <RegEx find="([sS])veštenic(a|e|u|om|i|ima)" replaceWith="$1većenic$2" />
+    <RegEx find="([sS])vež(a|e|u|om|oj)?" replaceWith="$1vjež$2" />
```

Comment:
Let's look at a trivial example:
```
[1] <RegEx find="a(b|c)?" replaceWith="x$1" />
```
Find an "a", optionally followed by "b" or "c", and replace the "a" with "x".
Because `(b|c)` is optional (`?` means zero or one), **every** "a" will be replaced by "x":
`aa ab ac ad a` ==> `xx xb xc xd x`
You will get the same result with `[1a] <RegEx find="a" replaceWith="x" />`.
```
[2] <RegEx find="a(b|c)?\b" replaceWith="x$1" />
```
After adding the word boundary `\b`, the transformation becomes:
`aa ab ac ad a` ==> `ax xb xc ad x`
The regular expression `a(b|c)?\b` only matches "a", "ab", or "ac" if they appear at the end of a word.
```
[3] <RegEx find="a(b|c)" replaceWith="x$1" />
```
Without the `?` the "b" or "c" becomes mandatory. An "a" will only be replaced with "x" if followed by "b" or "c":
`aa ab ac ad a` ==> `aa xb xc ad a`
_and also_ `abd ace acabf` ==> `xbd xce xcxbf` (not limited to the end of a word)

Both [2] and [3] could be useful transformations, but [1] is at best misleading (if there really is no need for limitation, [1a] should be used instead).

If you would really want to replace **every** "svež" with "svjež" (and "Svež" with "Svjež"), you would use
`<RegEx find="([sS])vež" replaceWith="$1vjež" />`
without the long, useless, misleading sub-expression.